spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/TestSparkOrcReadMetadataColumns.java

```diff
@@ -95,24 +99,20 @@ public class TestSparkOrcReadMetadataColumns {
     }
   }
 
-  @Parameterized.Parameters(name = "vectorized = {0}")
-  public static Object[] parameters() {
-    return new Object[] {false, true};
+  @Parameters(name = "vectorized = {0}")
+  public static Collection<Boolean> parameters() {
+    return Arrays.asList(false, true);
   }
 
-  @Rule public TemporaryFolder temp = new TemporaryFolder();
+  @TempDir private java.nio.file.Path temp;
 
-  private boolean vectorized;
+  @Parameter private boolean vectorized;
   private File testFile;
 
-  public TestSparkOrcReadMetadataColumns(boolean vectorized) {
-    this.vectorized = vectorized;
-  }
-
-  @Before
+  @BeforeEach
   public void writeFile() throws IOException {
-    testFile = temp.newFile();
-    Assert.assertTrue("Delete should succeed", testFile.delete());
+    testFile = File.createTempFile(UUID.randomUUID().toString(), null, temp.toFile());
```

Comment:
rather than generating a UUID in every single test it would be better to just use `junit` as the prefix. You could also use `Files.createTempFile(temp, "junit", ...)`, which takes a `Path` as a parameter, so you don't have to call `.toFile()`

Reply:
I tried a variation of that, but the tests weren't passing. Let me try this and I'll get back to you :)

Reply:
Removing the UUID.

The main problem with using `Files` is that we have another import 
`import org.apache.iceberg.Files;` that's being used.

So I end up writing
`java.nio.file.Files.createTempFile(temp, "junit", null).toFile();` 
which is not as readable as 
`File.createTempFile("junit", null, temp.toFile());`

Which one should I go ahead with?

Reply:
either is fine as long as it works ;)

Reply:
done!